graph/db/graph_cache.go

```diff
@@ -305,6 +410,9 @@ func (c *GraphCache) getChannels(node route.Vertex) []*DirectedChannel {
 		i++
 	}
 
+	// Copy the slice to clean up the unused pre allocated tail entries.
+	copy(channelsCopy, channelsCopy[:i])
```

Comment:
I think we can just return `channelsCopy[:i]` - in Go the re-slicing `[:i]` already creates a new slice header that points to the same underlying array as `channelsCopy`, but its length is `i`. And the copy will do nothing here.